lib/buffer.js

```diff
@@ -539,11 +539,29 @@ Buffer.prototype.inspect = function inspect() {
   return '<' + this.constructor.name + ' ' + str + '>';
 };
 
+function assertInteger(val, name) {
+  if (typeof val !== 'undefined' &&
+      (typeof val !== 'number' ||
+       +val !== (val >> 0))) {
```

Comment:
Easiest way to do this is `val != (val >>> 0)`. Though this also doesn't follow other Buffer APIs. We coerce almost everywhere else.